gitlab-ci/src/ci-scripts/docker-login.sh

```diff
@@ -0,0 +1,10 @@
+set -eEuo pipefail
+
+if which docker 2>/dev/null; then
+    docker login -u "$DOCKER_HUB_USER" -p "$DOCKER_HUB_PASSWORD_RO"
+elif which docker-bin 2>/dev/null; then
```

Comment:
If `docker` (podman) and `docker-bin` exist, we'll only login with `docker` but we should login with both (assuming the later is still needed). This change is not related to darwin IIUC. Is it needed? What for?

Reply:
Well before I figured out that the PATH var needed to be set it couldn't find the docker installation. So I this makes the error message clearer. But this goes back to having clearer documentation - it wasn't obvious to me why we were logging in twice.

Reply:
This (authentication with  `docker-bin`) might be used by docker rules. See `container_pull` in `WORKSPACES.bazel`. Just retain the logic as it was.